### PR TITLE
Test filter

### PR DESCRIPTION
**Pull request by nbhoski**
* Added support for filtering test cases based on folder and tag name
* two new options `select-by-folder` for filtering test folder & `select-by-tag` for filtering test tag is added.
* Updated the parameters and the description for newly added parameters.

### CODE REVIEW

**Comment by mcafaro** [comment]
Overall, looks good. Please modify the [integration tests](https://github.com/mathworks/matlab-circleci-orb/blob/e0c1630af6fcafa356eb0b5dcd91fc7bef0d4807/.circleci/config.yml#L88) to include test filtering.

File: src/commands/run-tests.yml [hunk 1]
```diff
@@ -23,6 +23,19 @@ parameters:
       semicolon-separated list.
     type: string
     default: ''
+  select-by-folder:
+    description: >
+      Location of the folder containing test code, relative to the project root folder. The
```
Comment (mw-hrastega) [inline src/commands/run-tests.yml:28]: How about this?

Location of the folder containing tests, relative to the project root folder. To generate a test suite, MATLAB uses only the tests in the specified folder and its subfolders. You can specify multiple folders using a colon-separated or a semicolon-separated list.

File: src/commands/run-tests.yml [hunk 2]
```diff
@@ -23,6 +23,19 @@ parameters:
...
+      specified folder and its subfolders are added to the test suite to run.To filter multiple test folders
+      You can specify multiple folders using a colon-separated or a semicolon-separated list.
+    type: string
+    default: ''
+  select-by-tag:
+    description: >
+      Specify the test Tag name. Test which are tagged with specified name will be added to test suite and
```
Comment (mw-hrastega) [inline src/commands/run-tests.yml:35]: Test tag used to select test suite elements. To generate a test suite, MATLAB uses only the test elements with the specified tag.

**Review by mw-hrastega — CHANGES_REQUESTED** [review]
I have some suggestions. Please let me know what you think.

File: src/commands/run-tests.yml (hunk 2 above)
Comment (nbhoski) [reply src/commands/run-tests.yml:35]: Sounds good to me. changed it

File: src/commands/run-tests.yml (hunk 1 above)
Comment (nbhoski) [reply src/commands/run-tests.yml:28]: Changed it.

**Comment by nbhoski** [comment]
@mcafaro added integ tests

File: .circleci/config.yml [hunk 3]
```diff
@@ -118,6 +131,16 @@ jobs:
...
+      - run:
+          name: Verify filtered test results file was created
+          command: |
+            set -e
+            grep -q simpleTest test-results/matlab/filterdtestresult.xml
+      - run:
+          name: Verify filter by tag test results file was created
+          command: |
+            set -e
+            grep -q TaggedTest test-results/matlab/filterdtagresult.xml
```
Comment (mcafaro) [inline .circleci/config.yml:143]: These tests should also confirm that the results **do not** contain a test that should be filtered.

**Review by mcafaro — CHANGES_REQUESTED** [review]

**Review by mw-hrastega — APPROVED** [review]

File: .circleci/config.yml (hunk 3 above)
Comment (nbhoski) [reply .circleci/config.yml]: Done.

File: src/commands/run-tests.yml [hunk 4]
```diff
@@ -23,6 +23,19 @@ parameters:
...
+      To generate a test suite, MATLAB uses only the tests in the specified folder and its subfolders.
+      You can specify multiple folders using a colon-separated or a semicolon-separated list.
+    type: string
+    default: ''
+  select-by-tag:
+    description: >
+      Test tag used to select test suite elements. To generate a test suite, MATLAB uses only the test elements
+      with the specified tag.
+    type: string
+    default: ''
```
Comment (mcafaro) [inline src/commands/run-tests.yml:38]: @mw-hrastega does it seem a bit odd we document these two "select-by" parameters in a different way? It seems like maybe there should be a common wording for all "select-by" parameters. Perhaps:

"Folder used to select test suite elements. To generate a ..."
"Test tag used to select test suite elements. To generate a..."

Comment (mw-hrastega) [reply src/commands/run-tests.yml:38]: I see your argument, Mark. Yes, it is a good idea to follow the same pattern for arguments belonging to the same category. However, we might want to mention that select-by-folder contains a path. Maybe we can say: 

Location of the folder used to select test suite elements, relative to the project root folder. To generate...

What do you think?

Comment (nbhoski) [reply src/commands/run-tests.yml]: I agree to @mw-hrastega  updated accordingly.

**Review by mw-hrastega — APPROVED** [review]

**Comment by mcafaro** [comment]
Overall looks good. There appears to be a build failure though.

**Pull request merged**